planner/core/plan.go

```diff
@@ -296,6 +299,16 @@ func (p *basePhysicalPlan) SetChildren(children ...PhysicalPlan) {
 	p.children = children
 }
 
+// SetChild implements LogicalPlan SetChild interface.
+func (p *baseLogicalPlan) SetChild(i int, child LogicalPlan) {
+	p.children[i] = child
+}
+
+// SetChild implements PhysicalPlan SetChild interface.
+func (p *basePhysicalPlan) SetChild(i int, child PhysicalPlan) {
```

Comment:
not needed?

Reply:
use in here: https://github.com/lzmhhh123/tidb/blob/8ed4d327cfacd7e002155868a28e58c26378825a/planner/core/rule_join_elimination.go#L230

It can avoid `make slice`.

Reply:
No. This function is binded to `PhysicalPlan`. But you only need a function binded to `LogicalPlan`.